lib/client/api.go

```diff
@@ -789,49 +789,79 @@ func (c *Config) SaveProfile(dir string, makeCurrent bool) error {
 	return nil
 }
 
-// ParseProxyHost parses the proxyHost string and updates the config.
+// ParsedProxyHost holds the hostname and Web & SSH proxy addresses
+// parsed out of a WebProxyAddress string.
+type ParsedProxyHost struct {
+	Host         string
+	WebProxyAddr string
+	SSHProxyAddr string
+}
+
+// ParseProxyHost parses a ProxyHost string of the format <hostname>:<proxy_web_port>,<proxy_ssh_port>
+// and returns the parsed components.
 //
-// Format of proxyHost string:
-//   proxy_web_addr:<proxy_web_port>,<proxy_ssh_port>
-func (c *Config) ParseProxyHost(proxyHost string) error {
+// If a definitive answer is not possible (e.g. no proxy port is specified in
+// the supplied string), the empty string is returned in the appropriate slot
+// of the resulting ParsedProxyHost.
+func ParseProxyHost(proxyHost string) (ParsedProxyHost, error) {
```

Comment:
Why not return a pointer here? Then this function would return `nil, trace.Wrap(err)` everywhere like usual. Does the caller use the return result even is there's an error?